giocomo_lab_to_nwb/wen22/wen21behaviorinterface.py

```diff
@@ -0,0 +1,97 @@
+from pathlib import Path
+from pynwb import NWBFile
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+from pynwb.behavior import Position, SpatialSeries
+from nwb_conversion_tools.tools.nwb_helpers import get_module
+from ndx_events import Events
+import pandas as pd
+
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from nwb_conversion_tools.utils.types import FolderPathType
+
+class Wen21EventsInterface(BaseDataInterface):
+    def __init__(self, session_path: FolderPathType):
+        super().__init__(session_path=session_path)
+
+    def run_conversion(
+        self,
+        nwb_file: NWBFile,
+        metadata: dict
+    ):
+
+        behavior_module = get_module(nwb_file, "behavior")
+        session_path = Path(self.source_data["session_path"])
+
+        # Add positions
+        position_file_path_list = list(session_path.glob("*position.txt"))
+        for position_file_path in position_file_path_list:
+            position_file_name = position_file_path.name
+            file_epoch_name = position_file_name.split("_position")[0].partition("train1")
+            if len(file_epoch_name) > 0:
+                file_epoch_name = file_epoch_name[1:]
+            df_data = pd.read_csv(
+                position_file_path,
+                sep="\t",
+                names=["position", "timestamps", "x1", "x2"],
+            )
+
+            pos_obj = Position(name=f"position during epoch {file_epoch_name[1:]}")
+
+            spatial_series_object = SpatialSeries(
+                name="position",
+                description="position within the virtual reality wheel",
+                data=H5DataIO(df_data.position.values, compression="gzip"),
+                reference_frame="unknown",
+                unit="cm",
+                conversion="0.100",
+                timestamps=df_data.timestamps.values,
+            )
+
+            pos_obj.add_spatial_series(spatial_series_object)
+            behavior_module.add_data_interface(pos_obj)
+
+        # Add lick events
+        lick_file_path_list = list(session_path.glob("*licks.txt"))
+        for lick_file_path in lick_file_path_list:
+
+            lick_file_name = lick_file_path.name
+            file_epoch_name = lick_file_name.split("_licks")[0].partition("train1")[-1]
+            if len(file_epoch_name) > 0:
+                file_epoch_name = file_epoch_name[1:]
+
+            licks_file_path = lick_file_path_list[0]
+            df_data = pd.read_csv(licks_file_path, sep="\t", names=["pos", "time"])
```

Comment:
Are we throwing out the position data then? That seems important. Maybe a `TimeSeries` inside a `BehavioralTimeSeries` would be better.

Reply:
Yes, I changed this. I was thinking that the information was redundant as 1) we also drop that information in the Mallory conversion and 2) the position time series (added in the conversion as well) contains a higher resolution version of the same information.

Reply:
Oh, I see, so there is Position over time, and lick instances, and if you take that lick time and look up the position, it's the same position?

Reply:
I think @h-mayorquin brought this up when I met with him - while it does count as 'redundant' information, there could be a performance benefit for analysis tools to having the position pre-extracted to correspond to each lick event. If someone wanted to grab it from the SpatialSeries, they would have to (i) do a lookup of the 'closest' timestamp of the SpatialSeries to that lick event time, (ii) use the index of that lookup to grab the values from the SpatialSeries data, which is far from the O(1) access provided by storing them side by side in this table and could take a while to do for all the lick events.

Reply:
Though *technically* if the (a) SpatialSeries is regular and (b) analysis software used on such a file is smart enough, it could infer the closest index through a O(1) calculation, so IDK - I'll leave it up to you to decide if you want to keep it or remove it from the table